llvm/lib/Target/X86/X86CallingConv.td

```diff
@@ -1056,6 +1056,22 @@ def CC_Intel_OCL_BI : CallingConv<[
   CCDelegateTo<CC_X86_32_C>
 ]>;
 
+def CC_X86_64_Preserve_None : CallingConv<[
+  // We don't preserve general registers, so all of them can be used to pass
+  // arguments except
+  //   - RBP  frame pointer
```

Comment:
It does not mention R16-R31

Reply:
Added a comment to mention that R16-R31 are not used to pass arguments because they are not universally available.